### PR TITLE
Check registered addons only used for sles migration test cases except sle-ha

### PR DESCRIPTION
**Pull request by tinawang123**
Check registered addons only used for sles migration test cases except sle-ha

- Related ticket: https://progress.opensuse.org/issues/54254
- Verification run: https://openqa.suse.de/tests/3257277#step/patch_sle/70

### CODE REVIEW

**Comment by mimi1vx** [comment]
Try verification run on released products, Your code before broke QAM SLE15+ jobs

**Comment by okurz** [comment]
s/Donot/Do not/ but please describe the reasoning for the change, not *what* you change.
Also see https://commit.style/ or http://chris.beams.io/posts/git-commit/ as a helpful guide how to write good commit messages.

**Comment by tinawang123** [comment]
Verified run:
12sp5 to 15sp1: https://openqa.suse.de/tests/3257121#step/zypper_lr/22
12sp4 to 12sp5: https://openqa.suse.de/tests/3257119#step/zypper_lr/10
opensuse: https://openqa.suse.de/tests/3257117#step/zypper_lr/2
sles4sap: https://openqa.suse.de/tests/3257233#step/patch_sle/32
With packagehub: https://openqa.suse.de/tests/3257241#step/zypper_lr/6

File: tests/update/patch_sle.pm [hunk 1]
```diff
@@ -48,11 +48,11 @@ sub patching_sle {
         set_scc_proxy_url if ((check_var('HDDVERSION', get_var('ORIGINAL_TARGET_VERSION')) && is_upgrade()));
         sle_register("register");
         zypper_call('lr -d');
-        # Check system version
-        if (get_var('ORIGIN_SYSTEM_VERSION')) {
+        # Check system version and registered addons
+        if (get_var('ORIGIN_SYSTEM_VERSION') and is_sle('<=15-sp1') and is_sle('12+')) {
```
Comment (coolgw) [inline tests/update/patch_sle.pm]: better add is_upgrade

**Comment by tinawang123** [comment]
Verified: https://openqa.suse.de/tests/3257277#step/zypper_lr/22

File: lib/registration.pm [hunk 2]
```diff
@@ -232,8 +232,10 @@ sub register_addons_cmd {
 
 sub check_registered_system {
     my ($system) = @_;
-    my $version = get_var('SLE_PRODUCT') . $system;
-    assert_script_run "zypper lr --uri | grep -i $version";
+    if (get_var('SLE_PRODUCT') eq 'sles') {
```
Comment (lemon-suse) [inline lib/registration.pm]: Why not use check_var?

**Comment by okurz** [comment]
In the git commit message: s/regiestered/registered/. Please take care to not introduce spelling mistakes and typos as correct english makes investigating in the git logs much easier, thanks!

**Comment by tinawang123** [comment]
> fix rest of former changes ...
Please help to review. I have changed the code.

**Comment by tinawang123** [comment]
Close this PR, as I will add this test code to our upgrade test module part. It will not impact any other part.

**Pull request closed without merging**